tensorly/tenalg/core_tenalg/n_mode_product.py

```diff
@@ -68,9 +69,31 @@ def mode_dot(tensor, matrix_or_vector, mode, transpose=False):
             f"Provided array of dimension {T.ndim(matrix_or_vector)} not in [1, 2]."
         )
 
-    res = T.dot(matrix_or_vector, unfold(tensor, mode))
+    unfolded = unfold(tensor, mode)
+    if tensor.dtype == torch.complex32 and not handle_complex_half_by_upcasting:
```

Comment:
Let's leave this for a separate PR, we need to think better on how to incorporate it cleanly. Also we should be using tl.dtype not directly torch.dtype since the latter will break on other backends.

Reply:
I agree with @JeanKossaifi, the support of complex operations in Tensorly, while interesting, should be handled separately.

Also, I believe some backends like Pytorch have built-in support for matrix multiplication with complex numbers, that may be faster than doing the splitting of real and imaginary parts manually, e.g. see [here](https://docs.pytorch.org/docs/stable/complex_numbers.html)